game-core/src/test/java/games/strategy/triplea/delegate/GameDataTestUtil.java

```diff
@@ -386,10 +397,22 @@ static BidPlaceDelegate bidPlaceDelegate(final GameData data) {
   }
 
   /**
-   * Returns a TestDelegateBridge for the given GameData + PlayerID objects.
+   * Returns a delegate bridge suitable for testing the given player and game data.
+   *
+   * @return A mock that can be configured using standard Mockito idioms.
    */
-  public static ITestDelegateBridge getDelegateBridge(final PlayerID player, final GameData data) {
-    return new TestDelegateBridge(data, player);
+  static ITestDelegateBridge getDelegateBridge(final PlayerID player, final GameData data) {
+    return spy(new TestDelegateBridge(data, player));
+  }
+
+  static OngoingStubbing<int[]> whenGetRandom(final IDelegateBridge delegateBridge) {
+    return when(delegateBridge.getRandom(anyInt(), anyInt(), any(), any(), anyString()));
```

Comment:
There are no assertions made over the passed parameters.
Given that those wouldn't be trivial to implement, do you think it's worth to extend the code to make assertions based on the passed array length?

Reply:
2c4e4dea7 adds an assertion to verify the actual count passed to `getRandom()` matches the expected count.  This actually revealed a few bugs in terms of what the expected rolls were supposed to be, so :+1: for that.

Another issue that arose is that, because we're now asserting the expected count within the mock, we have to setup the mock all at once instead of being able to spread it out for long tests (otherwise the assertion from a previous mock will run against the next call to `when()`).  In a sense, this is a good thing because the long tests are a smell.  Anyway, in order to provide context between the mocked rolls and where they are used, I ensured the comment next to the roll matches a similar comment in the test code that triggers the roll.  This was only required in a few tests, and it should be obvious where that was required while reviewing.